src/v3/signaling.ts

```diff
@@ -73,6 +81,10 @@ class Signaling extends EventEmitter {
     logger.debug("Disconnecting websocket");
     if (this.ws) {
       this.ws.notify("leave");
+      if (this.diagnosticsBatcher) {
```

Comment:
I don't know the details of the server side but just want to make sure that we don't need to call this before calling leave instead of after

Reply:
We discussed this at length, the reason for this ordering is we have an indeterminate amount of time to complete our best effort cleanup, and if we have to pick a call to succeed it's `leave` since that does more important cleanup work. Server side we'll just have to deal with it and make sure it doesn't cause any issues.

Reply:
Sounds good. Thanks for explaining